src/Java.Interop/Java.Interop/JniRuntime.JniValueManager.cs

```diff
@@ -590,6 +590,17 @@ public override void DestroyGenericArgumentState (IJavaPeerable value, ref JniVa
 		}
 
 		public override Expression CreateParameterFromManagedExpression (JniValueMarshalerContext context, ParameterExpression sourceValue, ParameterAttributes synchronize)
+		{
+			var r = CreateIntermediaryExpressionFromManagedExpression (context, sourceValue);
+
+			var h = Expression.Variable (typeof (IntPtr), sourceValue + "_handle");
```

Comment:
From the "wait, what was I thinking" department...  Is `sourceValue + "_handle"` *really* what we want?

Or should this instead be `sourceValue.Name + "_handle"`?

My guess is that `sourceValue + "_handle"` is a bug, and it "just happened" to work for us so far.

Reply:
Looks like `ParameterExpression:ToString ()` returns the `sourceValue.Name`, so it worked so far. I will change it to `sourceValue.Name + "_handle"` to be on the safe side. Which also means we would not need to update the tests hopefully :-)

Just to note, the documentation for the `ParameterExpression:ToString ()` states just: "Returns a textual representation of the Expression."